src/test/java/com/artipie/ResponseMetricsSliceTest.java

```diff
@@ -113,6 +122,57 @@ public void shouldReportBadAuthResponse() {
         );
     }
 
+    @Test
+    void shouldForwardRequestUnmodified() {
+        final String line = new RequestLine(RqMethod.POST, "/some_upload.war").toString();
+        final Header header = new Header("header1", "value1");
+        final byte[] body = "some code".getBytes();
+        new ResponseMetricsSlice(
+            (rqline, rqheaders, rqbody) -> {
+                MatcherAssert.assertThat(
+                    "Request line is forwarded as is",
+                    rqline,
+                    new IsEqual<>(line)
+                );
+                MatcherAssert.assertThat(
+                    "Headers are forwarded unmodified",
+                    rqheaders,
+                    Matchers.containsInAnyOrder(header)
+                );
+                MatcherAssert.assertThat(
+                    "Body is forwarded unmodified",
+                    new PublisherAs(rqbody).bytes().toCompletableFuture().join(),
+                    new IsEqual<>(body)
+                );
```

Comment:
@olenagerasimova I'd use `SliceHasResponse` + `RsHasStatus`, `RsHasHeaders` and `RsHasBody` matchers instead of checking it inside the callback, since it can be never called and test passes.